cs/src/core/expressions/TypeAlias.cs

```diff
@@ -111,8 +111,11 @@ static void GetSchemaTypes(Type type, HashSet<Type> schemaTypes)
         static Type GetConverter(Type type)
         {
             var name = type.AssemblyQualifiedName;
-            var converterName = type.Namespace + ".BondTypeAliasConverter" + name.Substring(type.FullName.Length);
-            return Type.GetType(converterName);
+            var converterName = type.Namespace + ".BondTypeAliasConverter";
+            // First try to locate converter class in the same assembly of the target type,
+            // If not found, try to locate in any assembly:
+            return Type.GetType(converterName + name.Substring(type.FullName.Length)) 
+                   ?? Type.GetType(converterName);
```

Comment:
At least looking at the documentation for [`Type.GetType(string)`](https://docs.microsoft.com/en-us/dotnet/api/system.type.gettype?view=netcore-3.1#System_Type_GetType_System_String_), a name without assembly qualification will only look in the currently executing assembly/mscorlib.dll/System.Private.CoreLib.dll.

Is something more subtle going on here that I'm not seeing?